packages/fe-fpm-writer/src/page/index.ts

```diff
@@ -0,0 +1,100 @@
+import { join } from 'path';
+import { create as createStorage } from 'mem-fs';
+import { create, Editor } from 'mem-fs-editor';
+import { render } from 'ejs';
+
+import { enhanceData } from './defaults';
+import { CustomPage, InternalCustomPage, Ui5Route } from './types';
+import { getTemplateRoot } from './version';
+
+/**
+ * Add a new route to the provided route array, and update existing routes if necessary (e.g. if targets are defined as arrays for FCL).
+ *
+ * @param routes existing application routes (from the manifest)
+ * @param config configuration object
+ */
+function updateRoutes(routes: Ui5Route[], config: InternalCustomPage) {
+    const newRoute: Partial<Ui5Route> = {
+        name: `${config.entity}${config.name}`
+    };
+    if (config.navigation) {
+        const sourceRoute = routes.find((route) => route.name === config.navigation?.sourcePage);
+        newRoute.pattern = `${sourceRoute?.pattern.replace(':?query:', '')}/${config.navigation.navEntity}({${
+            config.navigation.navEntity
+        }Key}):?query:`;
+        if (sourceRoute?.target.constructor === Array) {
+            const pages = sourceRoute.target as string[];
```

Comment:
sonarlint indicates that `as string` is superfluous

Reply:
df10458